### ISSUE TITLE
JSI support

### ISSUE THREAD

**Issue by Frans-L** [realm/realm-js#3815]
[JSI](https://github.com/react-native-community/discussions-and-proposals/issues/91) is one of the coming/new features of React Native.

Basically, JSI removes the bridge between JS and the Native side. The latest react native versions (e.g. 0.65) have "already" some JSI support. There are a couple of new libraries that are already using JSI and they are over 10x faster than their counterparts: e.g. [react-native-mmkv](https://github.com/mrousavy/react-native-mmkv) and [react-native-quick-sqlite](https://github.com/ospfranco/react-native-quick-sqlite).  However, Realm-JS seems to already call JavaScriptCore APIs directly, so JSI might not make so much difference. 

it seems like there are at least some planning about JSI in the Hermes PR draft: https://github.com/realm/realm-js/pull/3792.

Are there some official plans to start using JSI? Would JSI make Realm faster or is it needed just for Hermes support?

In other words, can we anticipate better perfomance in the future? :)

**Comment by kraenhansen** [comment]
We plan to support JavaScriptCore through our JSI implementation and remove our current direct implementation towards JavaScriptCore, primarily because it means less code to maintain. To my knowledge the primary performance benefits of JSI is related to the limited work the Hermes engine has to do on start-up and the lazy loading nature of turbo modules. Realm never used the RCT bridge since our API needed an ability to make synchronous operations when calling our C++ from JS.

Going forward Realm JS will only support debugging through the Safari dev-tools when running on JavaScriptCore or Flipper / Chrome dev-tools when running on Hermes. We expect this to significantly improve the debugging experience as we can finally ditch our current "RPC" debugging solution, which [has proven slow](https://github.com/realm/realm-js/issues/491) to the point of rendering apps unusable while debugging. Removing the support of the legacy Remote JS debugging via Chrome, seem to be the direction taken by other libraries in the community and we're looking forward to enable our community to use the far superior debugging experience that Flipper brings 🚀 

As such I'll close this issue, please follow our progress on #3792.
